packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -288,10 +304,30 @@ export const BuyMembershipForm = ({
 }
 
 export const BuyMembershipFormModal = ({ onClose, onSubmit, membershipPrice }: BuyMembershipFormModalProps) => {
+  const [buttonText, setButtonText] = useState<string | undefined>(undefined)
+  const handleSubmit = async (fields: MemberFormFields) => {
+    try {
+      if (fields.avatarUri && fields.avatarUri instanceof File) {
+        setButtonText('Uploading avatar...')
+        const data = await uploadAvatarImage(fields.avatarUri).then((res) => res.json())
+        onSubmit({ ...fields, avatarUri: `${process.env.REACT_APP_AVATAR_UPLOAD_URL}/${data.fileName}` })
+      } else {
+        onSubmit(fields)
+      }
+    } catch (e) {
+      onSubmit(fields)
+    }
+  }
```

Comment:
Again this logic is shared in a lot of places. WDYT of using a hook like:
```js
const useUploadAvatarAndSubmit = (onSubmit) => {
  const form = useFormContext()
  const [isUploading, setIsUploading] = useState(false)
  const uploadAvatarAndSubmit = useCallback(() => {
    setIsUploading(true)
    const avatarUri = await uploadAvatarImage(fields.avatarUri)
    onSubmit({ ...fields, avatarUri })
    setIsUploading(false)
  })
  return { isUploading, uploadAvatarAndSubmit }
}
```
With `uploadAvatarImage` changed too:
```ts
if (!avatarUri && avatarUri instanceof File) return avatarUri ?? ''
try {
  const body = new FormData()
  body.append('file', image, image.name)
  const data = fetch(process.env.REACT_APP_AVATAR_UPLOAD_URL ?? '', {
    method: 'POST',
    body,
  }).then((res) => res.json())
  return `${process.env.REACT_APP_AVATAR_UPLOAD_URL}/${data.fileName}`
} catch (err) {
  console.err('Error while uploading the avatar:', err)
  return ''
}
```